From: Stanislav Malyshev <stas@php.net>
Date: Sat Jul 28 22:16:29 2018 -0700
Subject: [PATCH] Fix for bug #76582

The brigade seems to end up in a messed up state if something fails
in shutdown, so we clean it up.

5.4 backport: https://github.com/shivammathur/php-src-backports/commit/a1c3f348b64ce5a9e61535aefeea460b72837562
---
diff --git a/sapi/apache2handler/sapi_apache2.c b/sapi/apache2handler/sapi_apache2.c
--- a/sapi/apache2handler/sapi_apache2.c
+++ b/sapi/apache2handler/sapi_apache2.c
@@ -678,6 +678,7 @@ zend_first_try {
 	if (!parent_req) {
 		php_apache_request_dtor(r TSRMLS_CC);
 		ctx->request_processed = 1;
+		apr_brigade_cleanup(brigade);
 		bucket = apr_bucket_eos_create(r->connection->bucket_alloc);
 		APR_BRIGADE_INSERT_TAIL(brigade, bucket);
 
